src/farcasterd/trade_state_machine.rs

```diff
@@ -27,8 +27,8 @@ use std::str::FromStr;
 /// ```ignore
 /// StartRestore    StartTaker  StartMaker
 ///       |             |            |
-///       |             |            V
-///       |             |        MakeOffer
+///       |             V            V
+///       |       TakerConnect   MakeOffer
```

Comment:
I suggest adding a line from `StartTaker` to `TakeOffer` as this transition can happen without the `TakerConnect` state.

Reply:
ed4247f160a06fd62682f44a36f16c0b1b149d55